library/core/src/intrinsics.rs

```diff
@@ -1747,8 +1748,11 @@ extern "rust-intrinsic" {
 
 /// Checks whether `ptr` is properly aligned with respect to
 /// `align_of::<T>()`.
-pub(crate) fn is_aligned_and_not_null<T>(ptr: *const T) -> bool {
-    !ptr.is_null() && ptr as usize % mem::align_of::<T>() == 0
+#[rustc_const_unstable(feature = "const_is_aligned_and_not_null", issue = "none")]
+pub(crate) const fn is_aligned_and_not_null<T>(ptr: *const T) -> bool {
+    use crate::ptr;
+    // FIXME: How would one check the alignment in a const way?
+    ptr_guaranteed_ne(ptr, ptr::null()) /*&& ptr.align_offset(mem::align_of::<T>()) == 0*/
```

Comment:
Instead of changing the function body here, I think it is better to just comment out the function calls like you did for `is_nonoverlapping`.

`align_offset` won't do the right thing in CTFE anyway, and once it does, we can just as well make the original code here working. So it's better to leave this code unchanged.